packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -151,16 +150,22 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
     private readonly contextsDeferred = new Map<string, Deferred<IChannelContext>>();
     private readonly pendingAttach = new Map<string, IAttachMessage>();
 
-    private bindState: BindState;
-    // For new data stores, this is used to break the recursion while attaching the graph. The graph must be attached
-    // before the data store can move to Attached state (see _attachState) and become live.
-    // For existing data stores, the graph is always attached.
-    private graphAttachState: AttachState = AttachState.Detached;
+    /**
+     * Tracks whether this data store is locally attached to the container's object graph. Being locally attached makes
+     * it visible in the container via the root of the graph.
+     * The data store must be locally attached before it can become live and be visible globally. In other words, if
+     * the data store is globally attached, it has to be locally attached as well.
+     */
+    private localAttachState: AttachState;
+    /**
+     * Tracks whether this data store is attached globally and is visible to all other clients. It must be locally
+     * attached before it can be globally attached.
+     */
+    private _globalAttachState: AttachState;
```

Comment:
Remove attaching state. I did not change it to a boolean because that will make it a breaking change as `attachState` is exposed on `IDataStoreRuntime`.
I can do that cleanup as a follow up to this change.

Reply:
Data store context also should not care about attaching state AFAIK. However, I am not touching the context in this change.

Reply:
I'd separate concerns of how something is exposed on public API (compat concerns) vs. how we track state internally - we can always to remapping on public APIs as a staging.
I do not like the need to have both localAttachState & _globalAttachState, I think it should be single enum covering all states.

Reply:
One more thing to add: If you want to use existing enum values, but know that you will not use one of them, you can do something like that:

private _globalAttachState: AttachState.Attached | AttachState.Detached;